perfkitbenchmarker/benchmarks/cloudsuite_web_serving.py

```diff
@@ -0,0 +1,411 @@
+  client.RemoteCommand('cd %s '
+                       '&& wget parsa.epfl.ch/cloudsuite/software/perfkit/'
+                       'web_serving/webservingfiles.tgz && '
+                       'tar xzf webservingfiles.tgz' % BASE_DIR)
+  client.RobustRemoteCommand('cd %s && '
+                             'cp -f %s'
+                             '/run.sh . && '
+                             'cp -f %s'
+                             '/run.xml . && '
+                             'cp -f %s'
+                             '/driver.policy . && '
+                             'cp -rf %s/build . && '
+                             'mkdir -p lib_olio && '
+                             'cp -rf %s/* lib_olio/ && '
+                             'cp -rf %s/resources build/ && '
+                             'export FABAN_HOME=%s  '
+                             % (FABAN_HOME, BASE_DIR, BASE_DIR, BASE_DIR,
+                                OLIO_BUILD, OLIO_BUILD, OLIO_BUILD,
+                                FABAN_HOME))
+  client.RemoteCommand('perl -pi -e '
+                       '"s/CLIENT_IP/%s/g"'
+                       ' %s/run.xml' % (CLIENT_IP, FABAN_HOME))
+  client.RemoteCommand('perl -pi -e '
+                       '"s/FRONTEND_IP/%s/g"'
+                       ' %s/run.xml' % (FRONTEND_IP, FABAN_HOME))
+  client.RemoteCommand('perl -pi -e '
+                       '"s/MYSQL_DIR/%s/g" '
+                       '%s/run.xml' % (re.escape(MYSQL_HOME), FABAN_HOME))
+  client.RemoteCommand('perl -pi -e '
+                       '"s/OUTPUT_DIR/%s/g" '
+                       '%s/run.xml' % (re.escape(OUTPUT_DIR), FABAN_HOME))
+  filestore = posixpath.join(frontend.GetScratchDir(), 'filestore')
+  client.RemoteCommand('perl -pi -e '
+                       '"s/FILESTORE_DIR/%s/g" '
+                       '%s/run.xml' % (re.escape(filestore), FABAN_HOME))
+  client.RemoteCommand('perl -pi -e '
+                       '"s/BACKEND_IP/%s/g" '
+                       '%s/run.xml' % (BACKEND_IP, FABAN_HOME))
+  return
+
+
+def GetInfo():
+  return BENCHMARK_INFO
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  if FLAGS['num_vms'].present and FLAGS.num_vms < 3:
+    raise ValueError('Web Serving requires at least 3 VMs')
+  return
+
+
+def PreparePrivateKey(vm):
+  vm.AuthenticateVm()
+
+
+def Prepare(benchmark_spec):
+  """Install Java, apache ant
+     Set up the client machine, backend machine, and frontend
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  vm_util.RunThreaded(PreparePrivateKey, vms)
+  client = vms[CL]
+  backend = vms[BK]
+  frontend = vms[FR]
+  CLIENT_IP = client.ip_address
+  for vm in vms:
+    vm.Install('wget')
+    vm.Install('ant')
+    vm.Install('openjdk7')
```

Comment:
Could these installations be moved into a parallel-executed function at the same time PreparePrivateKey is being called?